site/docs/skylark/windows_tips.md

```diff
@@ -0,0 +1,271 @@
+  relying less and less on Bash (MSYS2), so in the future users would be less likely to have MSYS2
+  installed along with Bazel. To make rules easier to use on Windows, avoid running Bash commands in
+  actions.
+
+- **Line endings**: Windows uses CRLF (`\r\n`), Unix-like systems uses LF (`\n`).
+
+  Be aware of this when comparing text files. Be mindful of your Git settings, especially of line
+  endings when checking out or committing. (See Git's `core.autocrlf` setting.)
+
+Solutions:
+
+- **Use a Bash-less purpose-made rule.**
+
+  `native.genrule()` is a wrapper for Bash commands, and it's often used to solve simple problems
+  like copying a file or writing a text file. You can avoid relying on Bash (and reinventing the
+  wheel): see if bazel-skylib has a purpose-made rule for your needs. None of them depends on Bash
+  when built/tested on Windows.
+
+  Build rule examples:
+
+  - `copy_file()`
+    ([source](https://github.com/bazelbuild/bazel-skylib/blob/master/rules/copy_file.bzl),
+    [documentation](https://github.com/bazelbuild/bazel-skylib/blob/master/docs/copy_file_doc.md)):
+    copies a file somewhere else, optionally making it executable
+
+  - `write_file()`
+    ([source](https://github.com/bazelbuild/bazel-skylib/blob/master/rules/write_file.bzl),
+    [documentation](https://github.com/bazelbuild/bazel-skylib/blob/master/docs/write_file_doc.md)):
+    writes a text file, with the desired line endings (`auto`, `unix`, or `windows`), optionally
+    making it executable (if it's a script)
+
+  - `run_binary()`
+    ([source](https://github.com/bazelbuild/bazel-skylib/blob/master/rules/run_binary.bzl),
+    [documentation](https://github.com/bazelbuild/bazel-skylib/blob/master/docs/run_binary_doc.md)):
+    runs a binary (or `*_binary` rule) with given inputs and expected outputs as a build action
+    (this is a build rule wrapper for `ctx.actions.run`)
+
+  - `native_binary()`
+    ([source](https://github.com/bazelbuild/bazel-skylib/blob/master/rules/native_binary.bzl),
+    [documentation](https://github.com/bazelbuild/bazel-skylib/blob/master/docs/native_binary_doc.md#native_binary)):
+    wraps a native binary in a `*_binary` rule, which you can `bazel run` or use in `run_binary()`'s
+    `tool` attribute or `native.genrule()`'s `tools` attribute
+
+  Test rule examples:
+
+  - `diff_test()`
+    ([source](https://github.com/bazelbuild/bazel-skylib/blob/master/rules/diff_test.bzl),
+    [documentation](https://github.com/bazelbuild/bazel-skylib/blob/master/docs/diff_test_doc.md)):
+    test that compares contents of two files
+
+  - `native_test()`
+    ([source](https://github.com/bazelbuild/bazel-skylib/blob/master/rules/native_binary.bzl),
+    [documentation](https://github.com/bazelbuild/bazel-skylib/blob/master/docs/native_binary_doc.md#native_test)):
+    wraps a native binary in a `*_test` rule, which you can `bazel test`
+
+- **On Windows, consider using `.bat` scripts for trivial things.**
```

Comment:
Maybe we could also mention using `powershell` as a solution?

Reply:
Good idea!

But I'm reluctant to do so. Running `.ps` scripts is harder than `.bat`: you can pass a `.bat` file to `CreateProcessW` directly (therefore also to `ctx.actions.run(executable={})`, but not a `.ps` file.

Reply:
I see, if we cannot run .ps directly when starting a process, then I think it's fine to only mention batch! Thanks!